<그룹>
- R*: 테스트베드를 통과하여 상용서비스가 가능한 알고리즘
- R+: 테스트베드 심사 중이거나 운용심사만 완료한 알고리즘
- R!: 관심 또는 주의가 필요한 알고리즘

<유형>
- 안정추구형, 위험중립형, 적극투자형

<용어>
- 표준편차: 수익률의 변동성을 나타내는 지표로 값이 클수록 변동성이 심하므로 위험이 크고, 값이 작을수록 위험이 작다고 할 수 있습니다.

- 샤프지수: 위험 1단위에 대한 초과수익의 정도를 나타내는 지표로써, 샤프지수가 높을수록 투자성과가 성공적이라고 할 수 있습니다. 단, 측정 기간과 투자 전략이 동일한 경우의 비교시에 의미가 있습니다.

- 최대손실률: 운용 기간 중 최고가를 기준으로 발생한 최대 손실의 비율을 표시합니다. 향후 투자자가 감내해야 할 위험의 수준을 알려주는 지표로 최대손실률이 적을 수록 안정적입니다.

- 보상비율: 최종손익을 최대손실폭으로 나눈 값으로 1미만인 경우 위험도에 비해 수익이 적고, 1이상인 경우 위험도에 비해 수익이 높고 안정적이라 볼 수 있습니다.

- 자산위험도: 자산군별 위험정도에 대한 값으로 0 ~ 10범위로 표시합니다. 값이 클수록 위험도가 높거나 투자에 적극적인 자산으로 구성되어 있음을 나타냅니다.
            자산군별 위험도의 값을 '주식 > 주식ETF > ELS/DLS > ETN > 펀드 > 기타상품 > 기타ETF > 채권ETF > RP' 순으로 설정하여 계산하였습니다.

- 유형차별성: 안정추구형, 위험중립형, 적극투자형 세가지 유형의 차별성을 나타내는 지표로 0 ~ 10 범위이며 값이 클수록 차별성이 높다고 할 수 있습니다.

- 종목편중률: 포트폴리오 구성종목 중 가장 높은 비중의 종목에 대한 자산군(색상)과 비율(%)을 표시합니다.(※ 예금,RP 등 현금성 자산은 제외합니다.)
            (※ 자산군 색상표시 :주식주식ETFELS/DLS ETN펀드기타상품 기타ETF채권ETF)

- 다계좌운용: 동일 유형 내 3개 계좌간 방향성의 차이를 나타낸 것으로, 값이 작을수록 다계좌 운용능력이 좋다고 할 수 있습니다. 다계좌운용지표는 현재 심사 중인 알고리즘에 대해서만 산출합니다.

- 자산별 투자비중: 자산군 분류기준(주식/펀드/주식ETF/채권ETF/기타ETF/ETN/예금/기타)에 따른 구분으로 자산 배분상태 확인이 가능합니다. RP는 예금으로, 해외ETF는 기타ETF로 분류됩니다.
            (※ 해외자산을 포함하는 포트폴리오의 운용실적은 환율변동에 의해 영향을 받을 수 있습니다.)

In [1]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
pd.set_option('display.max_columns', 27)
import re

url = "https://www.ratestbed.kr:7443/portal/pblntf/listProgrsInfo2.do?menuNo=200238"

res = requests.get(url)
soup = BeautifulSoup(res.text, "lxml")

date = soup.select_one('span.fc_point').text # 운용기준일
print(date)

2023년 07월 07일


In [2]:
columns_list = ['그룹', '알고리즘명', '유형', '업체명', '운용자금(원)',
                '1주', '1개월', '3개월', '6개월', '1년', '2년', '3년', '연환산', '누적',
                '표준편차', '샤프지수', '최대손실률', '보상비율',
                '자산 위험도', '유형 차별성', '종목 편중률', '편중된 종목', '다계좌 운용', '구성 종목수', '매매 회전률',
                '공시시작일', '공시기간(년)']

ra_df = pd.DataFrame(columns=columns_list)

ra_df

Empty DataFrame
Columns: [그룹, 알고리즘명, 유형, 업체명, 운용자금(원), 1주, 1개월, 3개월, 6개월, 1년, 2년, 3년, 연환산, 누적, 표준편차, 샤프지수, 최대손실률, 보상비율, 자산 위험도, 유형 차별성, 종목 편중률, 편중된 종목, 다계좌 운용, 구성 종목수, 매매 회전률, 공시시작일, 공시기간(년)]
Index: []

In [3]:
# 기본 정보
ra_df['그룹'] = [n.text for n in soup.select('p.name span.league_sym')] # 그룹
ra_df['알고리즘명'] = [n.text for n in soup.select('p.name span:nth-child(2)')] # 알고리즘명
ra_df['유형'] = [n.text for n in soup.select('p.clear span')] # 유형
ra_df['유형'] = ra_df['유형'].apply(lambda x: x.replace('[', '').replace(']', ''))
ra_df['업체명'] = [n.text for n in soup.select('td.logo span')] # 업체명
ra_df['운용자금(원)'] = [n.text.replace('\r', '').replace('\t', '').replace('\n', '').replace(',', '') for n in soup.select('td.amount')] # 운용자금(원)

# 수익률(%)
ra_df['1주'] = [n.text for n in soup.select('td.tar.profit1w')] # 수익률 1주
ra_df['1개월'] = [n.text for n in soup.select('td.tar.profit1m')] # 수익률 1달
ra_df['3개월'] = [n.text for n in soup.select('td.tar.profit3m')] # 수익률 3달
ra_df['6개월'] = [n.text for n in soup.select('td.tar.profit6m')] # 수익률 6달
ra_df['1년'] = [n.text for n in soup.select('td.tar.profit1y')] # 수익률 1년
ra_df['2년'] = [n.text for n in soup.select('td.tar.profit2y')] # 수익률 2년
ra_df['3년'] = [n.text for n in soup.select('td.tar.profit3y')] # 수익률 3년
ra_df['연환산'] = [n.text for n in soup.select('td.tar.profityc')] # 수익률 연환산
ra_df['누적'] = [n.text for n in soup.select('td.tar.profit')] # 수익률 누적

# 위험지표
ra_df['표준편차'] = [n.text for n in soup.select('td.tar.stdev')] # 표준편자
ra_df['샤프지수'] = [n.text for n in soup.select('td.tar.sharpe')] # 샤프지수
ra_df['최대손실률'] = [n.text for n in soup.select('td.tar.maxdd')] # 최대손실률
ra_df['보상비율'] = [n.text for n in soup.select('td.tar.reward')] # 보상비율

# 참고지표
ra_df['자산 위험도'] = [n.text for n in soup.select('td.tar.active')] # 자산위험도
ra_df['유형 차별성'] = [n.text for n in soup.select('td.tar.diff')] # 유형차별성
ra_df['종목 편중률'] = [n.text for n in soup.select('span.maxasset')] # 종목편중률
ra_df['편중된 종목'] = [n.attrs['title'] for n in soup.find_all('span', class_='maxasset')] # 편중된 종목
ra_df['다계좌 운용'] = [n.text.replace('\r', '').replace('\t', '').replace('\n', '') for n in soup.select('td.multiacc')] # 다계좌운용
ra_df['구성 종목수'] = [n.text for n in soup.select('td.tar.issuecnt')] # 구성종목수
ra_df['매매 회전률'] = [n.text for n in soup.select('td.tar.turnover')] # 매매회전율
ra_df['공시시작일'] = [n.text for n in soup.select('td[rowspan="1"]')] # 공시시작일
ra_df['공시기간(년)'] = [n.text.replace('년', '') for n in soup.select('span[class="optermText"]')] # 공시시간(년)

ra_df.head()

그룹                 알고리즘명                        유형     업체명  운용자금(원)     1주  \
0  *         콴텍 해외주식형 대형3호                     적극투자형  콴텍투자일임  2104405   1.37   
1  *         콴텍 해외주식형 대형3호                     안정추구형  콴텍투자일임  1877288  -1.05   
2  *         콴텍 해외주식형 대형3호                     위험중립형  콴텍투자일임  1967653  -0.28   
3  *  콴텍 밸류 Focus 국내 주식 1호  밸류 Focus 국내 주식 1호 적극투자형3  콴텍투자일임  3900000  -0.77   
4  *  콴텍 밸류 Focus 국내 주식 1호  밸류 Focus 국내 주식 1호 안정추구형3  콴텍투자일임  3900000  -0.41   

     1개월    3개월   6개월     1년     2년     3년    연환산      누적  표준편차   샤프지수  \
0   2.16   1.49  8.47  13.78  21.09  66.74   8.13   35.73  0.28   0.23   
1  -0.49  -2.39  4.82   3.33  12.15  13.04   3.69   13.56  0.09   0.21   
2   0.12  -1.42  4.27   4.57   7.13  19.13   3.44   13.11  0.14   0.13   
3   1.43   6.24  8.79   2.28                -9.53  -13.56  0.13  -0.02   
4   0.75   2.46  4.25   1.20                -4.75   -7.02  0.06  -0.00   

    최대손실률   보상비율 자산 위험도 유형 차별성 종목 편중률   편중된 종목 다계좌 운용 구성 종목수   매매 회전률  \
0  -43.09   0.77   7.51   6.03   9.37  해외기타ETF      -     11  1313.98   
1  -12.77   0.98   5.26   6.03  61.11  해외기타ETF      -     12   504.58   
2  -21.39   0.58   4.98   6.03  40.31  해외기타ETF      -     12   758.39   
3  -23.68  -0.57   8.66   6.77  11.20     국내주식      -     10   561.75   
4  -12.14  -0.58   4.47   6.77   5.88     국내주식      -     10   277.30   

        공시시작일 공시기간(년)  
0  2019-11-04     3.7  
1  2019-11-04     3.7  
2  2019-11-04     3.7  
3  2022-02-07     1.4  
4  2022-02-07     1.4

In [4]:
ra_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 683 entries, 0 to 682
Data columns (total 27 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   그룹       683 non-null    object
 1   알고리즘명    683 non-null    object
 2   유형       683 non-null    object
 3   업체명      683 non-null    object
 4   운용자금(원)  683 non-null    object
 5   1주       683 non-null    object
 6   1개월      683 non-null    object
 7   3개월      683 non-null    object
 8   6개월      683 non-null    object
 9   1년       683 non-null    object
 10  2년       683 non-null    object
 11  3년       683 non-null    object
 12  연환산      683 non-null    object
 13  누적       683 non-null    object
 14  표준편차     683 non-null    object
 15  샤프지수     683 non-null    object
 16  최대손실률    683 non-null    object
 17  보상비율     683 non-null    object
 18  자산 위험도   683 non-null    object
 19  유형 차별성   683 non-null    object
 20  종목 편중률   683 non-null    object
 21  편중된 종목   683 non-null    object
 22  다계

### 정렬하기 위해서 시리즈 타입을 숫자로 바꾸기

In [5]:
for column in ra_df.columns[4:21]:
    ra_df[column] = pd.to_numeric(ra_df[column], errors='coerce')

for column in ra_df.columns[22:25]:
    ra_df[column] = pd.to_numeric(ra_df[column], errors='coerce')

ra_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 683 entries, 0 to 682
Data columns (total 27 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   그룹       683 non-null    object 
 1   알고리즘명    683 non-null    object 
 2   유형       683 non-null    object 
 3   업체명      683 non-null    object 
 4   운용자금(원)  680 non-null    float64
 5   1주       680 non-null    float64
 6   1개월      680 non-null    float64
 7   3개월      512 non-null    float64
 8   6개월      512 non-null    float64
 9   1년       386 non-null    float64
 10  2년       237 non-null    float64
 11  3년       195 non-null    float64
 12  연환산      512 non-null    float64
 13  누적       680 non-null    float64
 14  표준편차     680 non-null    float64
 15  샤프지수     680 non-null    float64
 16  최대손실률    680 non-null    float64
 17  보상비율     680 non-null    float64
 18  자산 위험도   680 non-null    float64
 19  유형 차별성   680 non-null    float64
 20  종목 편중률   674 non-null    float64
 21  편중된 종목   683 non

In [6]:
ra_df.isnull().sum()

그룹           0
알고리즘명        0
유형           0
업체명          0
운용자금(원)      3
1주           3
1개월          3
3개월        171
6개월        171
1년         297
2년         446
3년         488
연환산        171
누적           3
표준편차         3
샤프지수         3
최대손실률        3
보상비율         3
자산 위험도       3
유형 차별성       3
종목 편중률       9
편중된 종목       0
다계좌 운용     494
구성 종목수       9
매매 회전률       6
공시시작일        0
공시기간(년)      0
dtype: int64

In [7]:
ra_df.head()

그룹                 알고리즘명                        유형     업체명    운용자금(원)    1주  \
0  *         콴텍 해외주식형 대형3호                     적극투자형  콴텍투자일임  2104405.0  1.37   
1  *         콴텍 해외주식형 대형3호                     안정추구형  콴텍투자일임  1877288.0 -1.05   
2  *         콴텍 해외주식형 대형3호                     위험중립형  콴텍투자일임  1967653.0 -0.28   
3  *  콴텍 밸류 Focus 국내 주식 1호  밸류 Focus 국내 주식 1호 적극투자형3  콴텍투자일임  3900000.0 -0.77   
4  *  콴텍 밸류 Focus 국내 주식 1호  밸류 Focus 국내 주식 1호 안정추구형3  콴텍투자일임  3900000.0 -0.41   

    1개월   3개월   6개월     1년     2년     3년   연환산     누적  표준편차  샤프지수  최대손실률  \
0  2.16  1.49  8.47  13.78  21.09  66.74  8.13  35.73  0.28  0.23 -43.09   
1 -0.49 -2.39  4.82   3.33  12.15  13.04  3.69  13.56  0.09  0.21 -12.77   
2  0.12 -1.42  4.27   4.57   7.13  19.13  3.44  13.11  0.14  0.13 -21.39   
3  1.43  6.24  8.79   2.28    NaN    NaN -9.53 -13.56  0.13 -0.02 -23.68   
4  0.75  2.46  4.25   1.20    NaN    NaN -4.75  -7.02  0.06 -0.00 -12.14   

   보상비율  자산 위험도  유형 차별성  종목 편중률   편중된 종목  다계좌 운용  구성 종목수   매매 회전률       공시시작일  \
0  0.77    7.51    6.03    9.37  해외기타ETF     NaN    11.0  1313.98  2019-11-04   
1  0.98    5.26    6.03   61.11  해외기타ETF     NaN    12.0   504.58  2019-11-04   
2  0.58    4.98    6.03   40.31  해외기타ETF     NaN    12.0   758.39  2019-11-04   
3 -0.57    8.66    6.77   11.20     국내주식     NaN    10.0   561.75  2022-02-07   
4 -0.58    4.47    6.77    5.88     국내주식     NaN    10.0   277.30  2022-02-07   

  공시기간(년)  
0     3.7  
1     3.7  
2     3.7  
3     1.4  
4     1.4

In [8]:
ra_df.describe()

운용자금(원)           1주         1개월         3개월         6개월  \
count  6.800000e+02   680.000000  680.000000  512.000000  512.000000   
mean   8.475034e+06     9.045824    1.155132    2.678164    8.717910   
std    1.131784e+07   155.008610   12.969221   14.308152   20.132392   
min    1.000000e+05    -6.360000  -32.820000  -13.850000  -10.410000   
25%    2.000000e+06    -0.890000   -0.430000   -0.637500    3.255000   
50%    5.000000e+06    -0.260000    0.360000    1.250000    5.750000   
75%    1.000000e+07     0.102500    1.330000    3.452500    9.597500   
max    6.500000e+07  3708.650000  242.490000  230.100000  321.170000   

               1년          2년          3년         연환산          누적        표준편차  \
count  386.000000  237.000000  195.000000  512.000000  680.000000  680.000000   
mean     3.963238   -1.400759   23.618974    2.432031    9.737750    0.152779   
std      7.252763   11.902142   29.278465   21.488286   31.312933    0.309242   
min    -17.530000  -37.660000   -3.770000 -249.540000  -33.960000    0.000000   
25%      0.262500   -7.940000    8.015000   -0.792500   -1.617500    0.060000   
50%      3.045000   -2.470000   13.700000    2.450000    1.695000    0.090000   
75%      6.705000    4.350000   26.990000    5.420000   13.857500    0.130000   
max     42.010000   35.910000  238.000000  109.360000  413.490000    3.290000   

             샤프지수       최대손실률        보상비율      자산 위험도      유형 차별성      종목 편중률  \
count  680.000000  680.000000  680.000000  680.000000  680.000000  674.000000   
mean     0.296971  -16.487265    0.468456    5.224368    6.820868   25.492329   
std      0.818929   18.133645    1.049070    1.943555    1.484740   17.436998   
min     -5.720000  -98.270000   -0.960000    0.000000    2.510000    2.540000   
25%     -0.000000  -19.617500   -0.200000    4.010000    5.800000   10.715000   
50%     -0.000000  -11.895000    0.210000    5.020000    7.040000   22.185000   
75%      0.320000   -5.375000    0.890000    6.180000    7.910000   34.287500   
max      5.010000    0.000000   10.170000    9.850000    9.870000   82.550000   

           다계좌 운용      구성 종목수       매매 회전률  
count  189.000000  674.000000   677.000000  
mean     0.112169   11.971810   529.632925  
std      0.186628    8.414456   600.598638  
min      0.000000    3.000000     3.450000  
25%      0.020000    7.000000   139.710000  
50%      0.060000   10.000000   327.970000  
75%      0.140000   13.000000   724.160000  
max      1.430000  101.000000  7663.690000

In [9]:
ra_df['그룹'].unique()

array(['*', '+', '!'], dtype=object)

### 파운트투자자문 R* 그룹 알고리즘 확인_6개

In [10]:
ra_df.loc[(ra_df['그룹'] == '*') & (ra_df['업체명'] == '파운트투자자문')].sort_values(by='누적', ascending=False)

그룹             알고리즘명     유형      업체명    운용자금(원)    1주   1개월   3개월   6개월  \
246  *   파운트 블루웨일(펀드)1.0  적극투자형  파운트투자자문   300000.0  1.10  4.52  3.85  5.67   
248  *   파운트 블루웨일(펀드)1.0  위험중립형  파운트투자자문   300000.0  0.94  3.86  3.26  4.78   
247  *   파운트 블루웨일(펀드)1.0  안정추구형  파운트투자자문   300000.0  0.61  2.84  2.75  3.94   
10   *  파운트 블루웨일(ETF)1.0  안정추구형  파운트투자자문  5000000.0 -1.59 -0.48 -2.89  2.41   
11   *  파운트 블루웨일(ETF)1.0  위험중립형  파운트투자자문  5000000.0 -1.36 -0.34 -2.86  2.30   
9    *  파운트 블루웨일(ETF)1.0  적극투자형  파운트투자자문  5000000.0 -1.44 -0.40 -3.29  0.93   

       1년     2년    3년   연환산     누적  표준편차  샤프지수  최대손실률  보상비율  자산 위험도  유형 차별성  \
246  0.85 -13.78  5.40  3.54  21.03  0.13  0.14 -22.30  0.65    5.70    7.11   
248  1.01 -11.23  3.52  3.01  17.64  0.15  0.09 -27.95  0.41    5.70    7.11   
247  2.53  -6.08  5.98  2.98  17.20  0.07  0.19 -11.70  1.16    5.70    7.11   
10  -0.80  -1.58 -2.28 -0.06  -1.51  0.08 -0.00 -12.90 -0.11    4.85    6.47   
11  -0.91  -3.55 -2.09 -0.24  -2.27  0.10 -0.00 -17.87 -0.12    4.73    6.47   
9   -2.06  -4.75 -2.87 -0.69  -4.52  0.14 -0.00 -28.06 -0.13    4.70    6.47   

     종목 편중률   편중된 종목  다계좌 운용  구성 종목수   매매 회전률       공시시작일 공시기간(년)  
246   35.15     국내펀드     NaN     5.0  1395.85  2018-05-08     5.2  
248   30.16     국내펀드     NaN     5.0  1297.44  2018-05-08     5.2  
247   29.73     국내펀드     NaN     5.0  1177.13  2018-05-08     5.2  
10    41.57  해외기타ETF     NaN     7.0  1457.66  2018-05-08     5.2  
11    31.61  해외기타ETF     NaN     8.0  1534.55  2018-05-08     5.2  
9     27.02  해외기타ETF     NaN     8.0  1601.13  2018-05-08     5.2

In [11]:
ra_df.loc[(ra_df['그룹'] == '*') & (ra_df['업체명'] == '파운트투자자문')].describe()

운용자금(원)        1주       1개월       3개월       6개월        1년  \
count  6.000000e+00  6.000000  6.000000  6.000000  6.000000  6.000000   
mean   2.650000e+06 -0.290000  1.666667  0.136667  3.338333  0.103333   
std    2.574296e+06  1.297104  2.333878  3.471493  1.767319  1.660791   
min    3.000000e+05 -1.590000 -0.480000 -3.290000  0.930000 -2.060000   
25%    3.000000e+05 -1.420000 -0.385000 -2.882500  2.327500 -0.882500   
50%    2.650000e+06 -0.375000  1.250000 -0.055000  3.175000  0.025000   
75%    5.000000e+06  0.857500  3.605000  3.132500  4.570000  0.970000   
max    5.000000e+06  1.100000  4.520000  3.850000  5.670000  2.530000   

              2년        3년       연환산         누적      표준편차      샤프지수  \
count   6.000000  6.000000  6.000000   6.000000  6.000000  6.000000   
mean   -6.828333  1.276667  1.423333   7.928333  0.111667  0.070000   
std     4.708976  4.131221  1.941862  11.832017  0.033116  0.082946   
min   -13.780000 -2.870000 -0.690000  -4.520000  0.070000 -0.000000   
25%    -9.942500 -2.232500 -0.195000  -2.080000  0.085000  0.000000   
50%    -5.415000  0.715000  1.460000   7.845000  0.115000  0.045000   
75%    -3.850000  4.930000  3.002500  17.530000  0.137500  0.127500   
max    -1.580000  5.980000  3.540000  21.030000  0.150000  0.190000   

           최대손실률      보상비율    자산 위험도    유형 차별성     종목 편중률  다계좌 운용    구성 종목수  \
count   6.000000  6.000000  6.000000  6.000000   6.000000     0.0  6.000000   
mean  -20.130000  0.310000  5.230000  6.790000  32.540000     NaN  6.333333   
std     7.172051  0.529717  0.517301  0.350542   5.162341     NaN  1.505545   
min   -28.060000 -0.130000  4.700000  6.470000  27.020000     NaN  5.000000   
25%   -26.537500 -0.117500  4.760000  6.470000  29.837500     NaN  5.000000   
50%   -20.085000  0.150000  5.275000  6.790000  30.885000     NaN  6.000000   
75%   -14.142500  0.590000  5.700000  7.110000  34.265000     NaN  7.750000   
max   -11.700000  1.160000  5.700000  7.110000  41.570000     NaN  8.000000   

            매매 회전률  
count     6.000000  
mean   1410.626667  
std     155.827412  
min    1177.130000  
25%    1322.042500  
50%    1426.755000  
75%    1515.327500  
max    1601.130000

### 디셈버앤컴퍼니(핀트) R* 그룹 알고리즘 확인_9개

In [12]:
ra_df.loc[(ra_df['그룹'] == '*') & (ra_df['업체명'] == '디셈버앤컴퍼니')].sort_values(by='누적', ascending=False)

그룹               알고리즘명            유형      업체명     운용자금(원)    1주   1개월  \
110  *       디셈버 미국 주식 솔루션  주식회사 디셈버앤컴퍼니  디셈버앤컴퍼니  45000000.0  0.25  1.46   
109  *       디셈버 미국 주식 솔루션  주식회사 디셈버앤컴퍼니  디셈버앤컴퍼니  45000000.0 -0.43  0.75   
272  *  디셈버 ISAAC 자산배분 해외형        적극투자형1  디셈버앤컴퍼니   5000000.0 -0.50  1.38   
271  *  디셈버 ISAAC 자산배분 해외형        위험중립형1  디셈버앤컴퍼니   5000000.0 -1.18  0.44   
108  *       디셈버 미국 주식 솔루션  주식회사 디셈버앤컴퍼니  디셈버앤컴퍼니  45000000.0 -1.12  0.02   
297  *  디셈버 ISAAC 자산배분 국내형        적극투자형1  디셈버앤컴퍼니   5000000.0 -0.09  1.04   
296  *  디셈버 ISAAC 자산배분 국내형        위험중립형1  디셈버앤컴퍼니   5000000.0 -0.23  0.53   
270  *  디셈버 ISAAC 자산배분 해외형        안정추구형1  디셈버앤컴퍼니   5000000.0 -1.82 -0.57   
295  *  디셈버 ISAAC 자산배분 국내형        안정추구형1  디셈버앤컴퍼니   5000000.0 -0.14  0.14   

      3개월    6개월     1년     2년     3년   연환산     누적  표준편차  샤프지수  최대손실률  보상비율  \
110  5.24  18.15  13.46  26.36  68.10  9.56  78.17  0.18  0.43 -34.33  1.89   
109  2.92  12.79   9.38  22.77  43.62  7.78  59.37  0.11  0.55 -21.50  2.32   
272  1.98  10.00   6.59   7.54  25.54  6.02  48.85  0.11  0.40 -21.90  1.77   
271  0.24   7.44   3.27   7.60  16.64  4.95  37.95  0.08  0.40 -16.16  1.94   
108  0.59   7.25   5.10  18.80  21.22  5.40  37.31  0.07  0.54 -11.81  2.21   
297  1.12   7.09   3.91  -3.12  13.31  3.60  26.83  0.12  0.17 -25.89  0.85   
296 -0.05   4.68   2.31  -2.21   8.02  2.82  20.52  0.07  0.16 -16.17  1.09   
270 -1.64   4.73   0.38   7.20   6.38  2.82  19.06  0.07  0.16 -11.59  1.30   
295 -0.46   2.67   2.33  -0.40   3.10  1.54  10.92  0.03 -0.00  -7.04  1.37   

     자산 위험도  유형 차별성  종목 편중률   편중된 종목  다계좌 운용  구성 종목수   매매 회전률       공시시작일  \
110    9.56    8.10    9.24  해외기타ETF     NaN    23.0  3511.64  2017-05-22   
109    7.71    8.10   23.08  해외기타ETF     NaN    26.0  2405.83  2017-05-22   
272    4.99    7.43   20.51  해외기타ETF     NaN    20.0  2235.22  2016-10-17   
271    4.99    7.43   12.91  해외기타ETF     NaN    22.0  1818.85  2016-10-17   
108    5.88    8.10   46.04  해외기타ETF     NaN    26.0  1294.15  2017-05-22   
297    3.97    7.68   19.84  국내기타ETF     NaN    16.0  2162.02  2016-10-17   
296    3.43    7.68   12.00  국내기타ETF     NaN    18.0  1680.49  2016-10-17   
270    4.99    7.43   37.60  해외기타ETF     NaN    16.0  1400.84  2016-10-17   
295    2.49    7.68   28.27  국내채권ETF     NaN    15.0  1321.06  2016-10-17   

    공시기간(년)  
110     6.1  
109     6.1  
272     6.7  
271     6.7  
108     6.1  
297     6.7  
296     6.7  
270     6.7  
295     6.7

In [13]:
ra_df.loc[(ra_df['그룹'] == '*') & (ra_df['업체명'] == '디셈버앤컴퍼니')].describe()

운용자금(원)        1주       1개월       3개월        6개월         1년  \
count  9.000000e+00  9.000000  9.000000  9.000000   9.000000   9.000000   
mean   1.833333e+07 -0.584444  0.576667  1.104444   8.311111   5.192222   
std    2.000000e+07  0.657934  0.662061  2.046033   4.755574   4.075107   
min    5.000000e+06 -1.820000 -0.570000 -1.640000   2.670000   0.380000   
25%    5.000000e+06 -1.120000  0.140000 -0.050000   4.730000   2.330000   
50%    5.000000e+06 -0.430000  0.530000  0.590000   7.250000   3.910000   
75%    4.500000e+07 -0.140000  1.040000  1.980000  10.000000   6.590000   
max    4.500000e+07  0.250000  1.460000  5.240000  18.150000  13.460000   

              2년         3년       연환산         누적      표준편차      샤프지수  \
count   9.000000   9.000000  9.000000   9.000000  9.000000  9.000000   
mean    9.393333  22.881111  4.943333  37.664444  0.093333  0.312222   
std    10.919479  20.915508  2.577120  21.534457  0.042720  0.194215   
min    -3.120000   3.100000  1.540000  10.920000  0.030000 -0.000000   
25%    -0.400000   8.020000  2.820000  20.520000  0.070000  0.160000   
50%     7.540000  16.640000  4.950000  37.310000  0.080000  0.400000   
75%    18.800000  25.540000  6.020000  48.850000  0.110000  0.430000   
max    26.360000  68.100000  9.560000  78.170000  0.180000  0.550000   

           최대손실률      보상비율    자산 위험도    유형 차별성     종목 편중률  다계좌 운용     구성 종목수  \
count   9.000000  9.000000  9.000000  9.000000   9.000000     0.0   9.000000   
mean  -18.487778  1.637778  5.334444  7.736667  23.276667     NaN  20.222222   
std     8.370334  0.508890  2.171740  0.293215  12.244506     NaN   4.265495   
min   -34.330000  0.850000  2.490000  7.430000   9.240000     NaN  15.000000   
25%   -21.900000  1.300000  3.970000  7.430000  12.910000     NaN  16.000000   
50%   -16.170000  1.770000  4.990000  7.680000  20.510000     NaN  20.000000   
75%   -11.810000  1.940000  5.880000  8.100000  28.270000     NaN  23.000000   
max    -7.040000  2.320000  9.560000  8.100000  46.040000     NaN  26.000000   

            매매 회전률  
count     9.000000  
mean   1981.122222  
std     705.443432  
min    1294.150000  
25%    1400.840000  
50%    1818.850000  
75%    2235.220000  
max    3511.640000

### DB 적재

In [14]:
import pymysql
import yaml
from sqlalchemy import create_engine, select, text
pymysql.install_as_MySQLdb()

# DB 접속 정보
with open('/mnt/FE0A5E240A5DDA6B/workspace/Quant_Portfolio/Analyst/config.yaml', encoding='UTF-8') as f:
    _cfg = yaml.load(f, Loader=yaml.FullLoader)

DB_SECRET = _cfg['DB_SECRET']

# MySQL에 DF 밀어넣는 함수
def df2sql(is_data: pd.DataFrame, db_table):

    engine = create_engine(f"mysql+pymysql://root:{DB_SECRET}@localhost:3306/quant_db?charset=utf8", encoding='utf8')
    conn = engine.connect()

    is_data.to_sql(name=db_table, con=engine, if_exists='replace', index=False) # if_exists='append'
    conn.close()

df2sql(ra_df, 'tb_RAtestbed')

### sqlalchemy로 DB 불러오기 시도

In [15]:
engine = create_engine(f"mysql+pymysql://root:{DB_SECRET}@localhost:3306/quant_db?charset=utf8", encoding='utf8')

with engine.connect() as conn:
    result = conn.execute(text("SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '파운트투자자문';"))
    for row in result:
        print(row) # <class 'sqlalchemy.engine.row.LegacyRow'>

('*', '파운트 블루웨일(ETF)1.0', '적극투자형', '파운트투자자문', 5000000.0, -1.44, -0.4, -3.29, 0.93, -2.06, -4.75, -2.87, -0.69, -4.52, 0.14, -0.0, -28.06, -0.13, 4.7, 6.47, 27.02, '해외기타ETF', None, 8.0, 1601.13, '2018-05-08', '5.2')
('*', '파운트 블루웨일(ETF)1.0', '안정추구형', '파운트투자자문', 5000000.0, -1.59, -0.48, -2.89, 2.41, -0.8, -1.58, -2.28, -0.06, -1.51, 0.08, -0.0, -12.9, -0.11, 4.85, 6.47, 41.57, '해외기타ETF', None, 7.0, 1457.66, '2018-05-08', '5.2')
('*', '파운트 블루웨일(ETF)1.0', '위험중립형', '파운트투자자문', 5000000.0, -1.36, -0.34, -2.86, 2.3, -0.91, -3.55, -2.09, -0.24, -2.27, 0.1, -0.0, -17.87, -0.12, 4.73, 6.47, 31.61, '해외기타ETF', None, 8.0, 1534.55, '2018-05-08', '5.2')
('*', '파운트 블루웨일(펀드)1.0', '적극투자형', '파운트투자자문', 300000.0, 1.1, 4.52, 3.85, 5.67, 0.85, -13.78, 5.4, 3.54, 21.03, 0.13, 0.14, -22.3, 0.65, 5.7, 7.11, 35.15, '국내펀드', None, 5.0, 1395.85, '2018-05-08', '5.2')
('*', '파운트 블루웨일(펀드)1.0', '안정추구형', '파운트투자자문', 300000.0, 0.61, 2.84, 2.75, 3.94, 2.53, -6.08, 5.98, 2.98, 17.2, 0.07, 0.19, -11.7, 1.16, 5.7, 7.11, 29.73, 

In [16]:
# Ver. 1: failed
# stmt = select(tb_RAtestbed).where(tb_RAtestbed.그룹 == '*')
# print(stmt)

# Ver. 2
engine = create_engine(f"mysql+pymysql://root:{DB_SECRET}@localhost:3306/quant_db?charset=utf8", encoding='utf8')
conn = engine.connect()

ra_df_from_sql = pd.read_sql_table('tb_RAtestbed', conn)
ra_df_from_sql.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 683 entries, 0 to 682
Data columns (total 27 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   그룹       683 non-null    object 
 1   알고리즘명    683 non-null    object 
 2   유형       683 non-null    object 
 3   업체명      683 non-null    object 
 4   운용자금(원)  680 non-null    float64
 5   1주       680 non-null    float64
 6   1개월      680 non-null    float64
 7   3개월      512 non-null    float64
 8   6개월      512 non-null    float64
 9   1년       386 non-null    float64
 10  2년       237 non-null    float64
 11  3년       195 non-null    float64
 12  연환산      512 non-null    float64
 13  누적       680 non-null    float64
 14  표준편차     680 non-null    float64
 15  샤프지수     680 non-null    float64
 16  최대손실률    680 non-null    float64
 17  보상비율     680 non-null    float64
 18  자산 위험도   680 non-null    float64
 19  유형 차별성   680 non-null    float64
 20  종목 편중률   674 non-null    float64
 21  편중된 종목   683 non

### 디셈버앤컴퍼니(핀트) 누적 수익률 확인: 37.7%/ 6.1년, 6.7년

In [18]:
import pandas as pd
pd.set_option('display.max_columns', 27)

conn = pymysql.connect(host='localhost', user='root',
    password=DB_SECRET, db='quant_db', charset='utf8')

SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '디셈버앤컴퍼니';"

ra_df_from_sql = pd.read_sql(SQL, conn)

ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간(년)'].unique()

/tmp/ipykernel_91596/4192346678.py:9: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(37.66444444444445, array(['6.1', '6.7'], dtype=object))

### 쿼터백자산운용 누적 수익률 확인: 32.2% / 6.7년

In [19]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '쿼터백자산운용';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간(년)'].unique()

/tmp/ipykernel_91596/3673429839.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(32.215, array(['6.7'], dtype=object))

### 콴텍 누적 수익률 확인: 14.4% / 0.7년, 0.9년, 1.4년, 1.7년, 2.4년, 3.7년, 4.4년

In [20]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '콴텍투자일임';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간(년)'].unique()

/tmp/ipykernel_91596/1780006522.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(14.444080459770115,
 array(['3.7', '1.4', '0.9', '0.7', '2.4', '4.4', '1.7'], dtype=object))

### 파운트 누적 수익률 확인: 7.9% / 5.2년

In [21]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '파운트투자자문';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간(년)'].unique()

/tmp/ipykernel_91596/2010170521.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(7.928333333333334, array(['5.2'], dtype=object))

### 업라이즈투자자문(이루다) 누적 수익률 확인: 5% / 1.7년, 4.4년, 6.7년

In [22]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '업라이즈투자자문';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간(년)'].unique()

/tmp/ipykernel_91596/803094738.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(5.032, array(['1.7', '4.4', '6.7'], dtype=object))

### 퀀팃투자자문 누적 수익률 확인: 3.1% / 2.2년

In [23]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '퀀팃투자자문';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간(년)'].unique()

/tmp/ipykernel_91596/3191469704.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(3.11, array(['2.2'], dtype=object))

### 크래프트테크놀로지스(&하나은행) 누적 수익률 확인: 2.6% / 6.1년

In [24]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 알고리즘명 like '%크래프트%';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간(년)'].unique()

/tmp/ipykernel_91596/3565763615.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(2.6166666666666667, array(['6.1'], dtype=object))

### Query로 RA 투자일임 회사 비교하는 DataFrame 만들기

In [25]:
engine = create_engine(f"mysql+pymysql://root:{DB_SECRET}@localhost:3306/quant_db?charset=utf8", encoding='utf8')

with engine.connect() as conn:
    result = conn.execute(text("SELECT 업체명, 유형, \
                               ROUND(AVG(1개월), 1), ROUND(AVG(3개월), 1), ROUND(AVG(6개월), 1), \
                               ROUND(AVG(1년), 1), ROUND(AVG(2년), 1), ROUND(AVG(3년), 1), \
                               ROUND(AVG(연환산), 1), ROUND(AVG(누적), 1), \
                               ROUND(AVG(샤프지수), 1), ROUND(AVG(최대손실률), 1), \
                               ROUND(AVG(보상비율), 1), ROUND(AVG(자산 위험도), 1), \
                               ROUND(AVG(종목 편중률), 1), ROUND(AVG(구성 종목수), 1), ROUND(AVG(매매 회전율), 1), ROUND(AVG(공시기간), 1) \
                               FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' GROUP BY 업체명 ORDER BY AVG(누적) DESC;"))
    for row in result:
        print(row)

ProgrammingError: (pymysql.err.ProgrammingError) (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '위험도), 1),                                ROUND(AVG(종목 편중률), 1), ' at line 1")
[SQL: SELECT 업체명, 유형,                                ROUND(AVG(1개월), 1), ROUND(AVG(3개월), 1), ROUND(AVG(6개월), 1),                                ROUND(AVG(1년), 1), ROUND(AVG(2년), 1), ROUND(AVG(3년), 1),                                ROUND(AVG(연환산), 1), ROUND(AVG(누적), 1),                                ROUND(AVG(샤프지수), 1), ROUND(AVG(최대손실률), 1),                                ROUND(AVG(보상비율), 1), ROUND(AVG(자산 위험도), 1),                                ROUND(AVG(종목 편중률), 1), ROUND(AVG(구성 종목수), 1), ROUND(AVG(매매 회전율), 1), ROUND(AVG(공시기간), 1)                                FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' GROUP BY 업체명 ORDER BY AVG(누적) DESC;]
(Background on this error at: https://sqlalche.me/e/14/f405)

In [ ]:
columns = ['업체명', '1주', '1개월', '3개월', '6개월', '1년',
       '2년', '3년', '연환산', '누적', '표준편차', '샤프지수', '최대손실률', '보상비율', '자산 위험도',
       '유형 차별성', '종목 편중률', '다계좌 운용', '구성 종목수', '매매 회전률']

Rstar_versus_df = pd.DataFrame(columns=columns)

Rstar_versus_df['업체명'] = ['디셈버앤컴퍼니', '쿼터백자산운용', '콴텍투자일임', '파운트투자자문', '업라이즈투자자문', '퀀팃투자자문', '%크래프트%']

for 

engine = create_engine(f"mysql+pymysql://root:{DB_SECRET}@localhost:3306/quant_db?charset=utf8", encoding='utf8')

with engine.connect() as conn:
    result = conn.execute(text("SELECT 업체명, ROUND(AVG(누적), 1) FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' GROUP BY 업체명 ORDER BY AVG(누적) DESC;"))
    for row in result:
        print(row)

# 코스피 vs. 성장가치 소형주 with 한계기업 vs. 성장가치 소형주 without 한계기업 수익률 비교

In [1]:
import yfinance as yf
import pandas as pd

kospi = yf.download('^KS11', start='2023-01-01', end='2023-07-01', progress=True, auto_adjust=True) # kospi200: ^KS200, kospi: ^KS11
# kospi.head()

(kospi.iloc[-1, 0] * 100 / kospi.iloc[0, 3]) - 100

[*********************100%***********************]  1 of 1 completed


14.95100448162114

In [109]:
kospi.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 863 entries, 2020-01-02 to 2023-06-30
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Open    863 non-null    float64
 1   High    863 non-null    float64
 2   Low     863 non-null    float64
 3   Close   863 non-null    float64
 4   Volume  863 non-null    int64  
dtypes: float64(4), int64(1)
memory usage: 40.5 KB


In [110]:
kospi.tail()

Open         High          Low        Close  Volume
Date                                                                  
2023-06-26  2568.649902  2585.310059  2562.030029  2582.199951  479200
2023-06-27  2581.020020  2582.000000  2568.520020  2581.389893  535400
2023-06-28  2590.520020  2590.520020  2559.459961  2564.189941  484200
2023-06-29  2564.020020  2577.689941  2550.020020  2550.020020  602500
2023-06-30  2558.429932  2571.030029  2541.719971  2564.280029  508800

In [76]:
import yfinance as yf
import pandas as pd

kosdaq = yf.download('^KQ11', start='2023-01-01', end='2023-07-01', progress=True, auto_adjust=True) # kosdaq: ^KQ11
# kosdaq.head()

(kosdaq.iloc[-1, 0] * 100 / kosdaq.iloc[0, 3]) - 100

[*********************100%***********************]  1 of 1 completed


28.471653387287063

In [96]:
kosdaq.tail()

Open        High         Low       Close  Volume
Date                                                              
2023-06-26  874.309998  881.239990  869.289978  879.500000    1300
2023-06-27  877.880005  878.429993  867.679993  874.140015    1200
2023-06-28  879.609985  880.229980  866.700012  866.969971    1200
2023-06-29  868.650024  870.750000  859.770020  861.789978    1200
2023-06-30  862.700012  869.239990  853.679993  868.239990    1000

In [73]:
# 한계기업 포함 Top 10

top10_withrisk = ['010770',
 '019540',
 '098660',
 '031510',
 '001210',
 '227100',
 '044480',
 '052300',
 '091440',
 '084870']

In [72]:
# 한계기업 제외 Top 10

top10_withoutrisk = ['021650',
 '031310',
 '046310',
 '032750',
 '006920',
 '001020',
 '290270',
 '012620',
 '203450',
 '051390']

In [113]:
import pandas_datareader.data as web

earning_total_withrisk = 0

for code in top10_withrisk:

    df = web.DataReader(
        code,
        'naver',
        start='20200101',
        end='20230630'
    )

    earning_total_withrisk += (int(df.iloc[-1, 0]) * 100 / int(df.iloc[0, 3])) - 100

earning_total_withrisk = int(earning_total_withrisk) / len(top10_withrisk)

earning_total_withrisk

-35.4

In [114]:
import pandas_datareader.data as web

earning_total_withoutrisk = 0

for code in top10_withoutrisk:

    df = web.DataReader(
        code,
        'naver',
        start='20200101',
        end='20230630'
    )

    earning_total_withoutrisk += (int(df.iloc[-1, 0]) * 100 / int(df.iloc[0, 3])) - 100

earning_total_withoutrisk = int(earning_total_withoutrisk) / len(top10_withoutrisk)

earning_total_withoutrisk

-1.5

# 2022 코스피 수익률 -24.2% / 2023 14.9% / 20 ~ 23.6.30 17.6%

# 2022 성장가치 소형주 & 한계기업 수익률 -38.6% / 2023 6.3% / 20 ~ 23.6.30 -35.4%

# 2022 성장가치 소형주 without 한계기업 수익률 -29.4 / 2023 13.1% / 20 ~ 23.6.30 -1.5%

In [107]:
13.1 * 100 / 6.3

207.93650793650795

In [117]:
(13.1 * 100) / 14.9 - 100

-12.080536912751683